### PR TITLE
feat: fill shippingInfo on setShippingMethod

### PR DESCRIPTION
**Pull request by mickdekkers**
We were having some trouble writing tests around a shipping method because `cart.shippingInfo.taxedPrice` wasn't populated by the mocking library. This PR aims to fix that.

## Status

Currently looking for feedback, particularly on the unresolved questions below. A couple t's left to cross and i's left to dot, but so far the newly written tests pass and it works as expected when testing locally (symlinked) in our project.

Will also add a changeset.

## Questions

There's a couple remaining questions around edge cases:

- https://github.com/mickdekkers/commercetools-node-mock/blob/9bb75efa355dd6e5c3fa409b98125b362e08bf8b/src/repositories/cart/actions.ts#L654
  - I think the note in the `zone.location` field's [docs](https://docs.commercetools.com/api/projects/zones#location) answers this, can anyone confirm?
- https://github.com/mickdekkers/commercetools-node-mock/blob/9bb75efa355dd6e5c3fa409b98125b362e08bf8b/src/repositories/cart/actions.ts#L665-L666
  - The ZoneRate [docs](https://docs.commercetools.com/api/projects/shippingMethods#zonerate) mention the shipping rates are defined by currency, but can anyone confirm is this the only differentiating factor? If so, there should be exactly one entry in the array after `getShippingMethodsMatchingCart` processes it, so `[0]` is correct
- https://github.com/mickdekkers/commercetools-node-mock/blob/0f63c19ee10374fc97624f7812a9d9665ed7e2df/src/services/cart.test.ts#L1111
  - It's unclear from [the docs](https://docs.commercetools.com/api/projects/carts#shippinginfo) whether `cart.shippingInfo.price` should be the net or gross price.

### CODE REVIEW

File: src/shipping.ts [hunk 1]
```diff
@@ -0,0 +1,147 @@
...
+	}
+
+	const tierType = tiers[0].type;
+	switch (tierType) {
+		case "CartValue":
+			return markMatchingCartValueTiers(cart, tiers as CartValueTier[]);
+		// case 'CartClassification':
+		// 	return markMatchingCartClassificationTiers(cart, tiers)
+		// case 'CartScore':
+		// 	return markMatchingCartScoreTiers(cart, tiers)
```
Comment (okkevandereijk) [inline src/shipping.ts:44]: Why is this commented out?

**Review by okkevandereijk — APPROVED** [review]

File: src/shipping.ts (hunk 1 above)
Comment (mickdekkers) [reply src/shipping.ts:44]: I don't know. This was part of a file that was moved: `src/shippingCalculator.ts`

**Review by mickdekkers — COMMENTED** [review]
My bad, I still had the comment on line 12 of `src/shipping.ts` in draft. Thought I'd submitted it already
